libs/application-generic/src/usecases/get-preferences/get-preferences.usecase.ts

```diff
@@ -112,33 +76,108 @@ export class GetPreferences {
     }
   }
 
-  private getSubscriberWorkflowPreferences(
-    items: PreferencesEntity[],
-    templateId: string,
-  ) {
-    return items.find(
-      (item) =>
-        item.type === PreferencesTypeEnum.SUBSCRIBER_WORKFLOW &&
-        item._templateId === templateId,
+  /** Transform WorkflowOptionsPreferences into IPreferenceChannels */
+  public static mapWorkflowOptionsPreferencesToChannels(
+    workflowPreferences: WorkflowOptionsPreferences
+  ): IPreferenceChannels | undefined {
+    if (!workflowPreferences) {
+      return undefined;
+    }
+
+    return {
+      in_app: workflowPreferences.channels.in_app.defaultValue || workflowPreferences.workflow.defaultValue,
+      sms: workflowPreferences.channels.sms.defaultValue || workflowPreferences.workflow.defaultValue,
+      email: workflowPreferences.channels.email.defaultValue || workflowPreferences.workflow.defaultValue,
+      push: workflowPreferences.channels.push.defaultValue || workflowPreferences.workflow.defaultValue,
+      chat: workflowPreferences.channels.chat.defaultValue || workflowPreferences.workflow.defaultValue,
+    };
+  }
+
+  /** Determine if Workflow Preferences should be marked as critical / readOnly at the top level */
+  public static checkIfWorkflowPreferencesIsReadOnly(workflowPreferences?: WorkflowOptionsPreferences): boolean {
+    if (!workflowPreferences) {
+      return false;
+    }
+
+    return (
+      workflowPreferences.workflow.readOnly ||
+      Object.values(workflowPreferences.channels).some(({ readOnly }) => readOnly)
     );
   }
 
-  private getSubscriberGlobalPreferences(items: PreferencesEntity[]) {
+  private mergePreferences(items: PreferencesEntity[], workflowId?: string): WorkflowChannelPreferences | undefined {
+    const workflowPreferences = this.getWorkflowPreferences(items);
+
+    const userPreferences = this.getUserPreferences(items);
+
+    const subscriberGlobalPreferences = this.getSubscriberGlobalPreferences(items);
+
+    const subscriberWorkflowPreferences = this.getSubscriberWorkflowPreferences(items, workflowId);
+
+    /**
+     * Order is important here because we like the workflowPreferences (that comes from the bridge)
+     * to be overridden by any other preferences and then we have preferences defined in dashboard and
+     * then subscribers global preferences and the once that should be used if it says other then anything before it
+     * we use subscribers workflow preferences
+     */
+    const preferences = [
+      workflowPreferences,
+      userPreferences,
+      subscriberGlobalPreferences,
+      subscriberWorkflowPreferences,
+    ]
+      .filter((preference) => preference !== undefined)
+      .map((item) => item.preferences);
+
+    // ensure we don't merge on an empty list
+    if (preferences.length === 0) {
+      return;
+    }
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** This is existing functionality just moved‏